compio-dispatcher/src/lib.rs

```diff
@@ -6,28 +6,37 @@ use std::{
     future::Future,
     io,
     num::NonZeroUsize,
-    panic::resume_unwind,
+    panic::{resume_unwind, UnwindSafe},
+    sync::{Arc, Mutex},
     thread::{available_parallelism, JoinHandle},
 };
 
-use crossbeam_channel::{unbounded, SendError, Sender};
+use compio_driver::{AsyncifyPool, ProactorBuilder};
+use compio_runtime::event::{Event, EventHandle};
+use crossbeam_channel::{unbounded, Sender};
 use futures_util::{future::LocalBoxFuture, FutureExt};
 
-type BoxClosure<'a> = Box<dyn (FnOnce() -> LocalBoxFuture<'a, io::Result<()>>) + Send>;
-
 /// The dispatcher. It manages the threads and dispatches the tasks.
 pub struct Dispatcher {
-    sender: Sender<BoxClosure<'static>>,
-    threads: Vec<JoinHandle<io::Result<()>>>,
+    sender: Sender<DispatcherClosure>,
+    threads: Vec<JoinHandle<()>>,
+    pool: Arc<AsyncifyPool>,
 }
 
 impl Dispatcher {
     /// Create the dispatcher with specified number of threads.
     pub(crate) fn new_impl(mut builder: DispatcherBuilder) -> io::Result<Self> {
-        let (sender, receiver) = unbounded::<BoxClosure<'static>>();
+        let mut proactor_builder = builder.proactor_builder;
+        let pool = proactor_builder.create_or_get_thread_pool();
```

Comment:
Set the pool in `create_or_get_thread_pool`

Reply:
A build method should not modify the builder itself.

Reply:
Why is that? The whole meaning of build method is to alternate the builder.

Reply:
When a user doesn't specify to reuse the pool, we should not reuse it.